app/main.py

```diff
@@ -36,17 +36,37 @@ async def upload_files(request: Request, session: SessionDep):
     files = body["files"]
     filesMimes = body["filesMimes"]
     filesNames = body["filesNames"]
-    file_result: dict[str, MarcFile] = {}
+    file_result: dict[str, IncompleteFile] = {}
 
     for file, mime, name in zip(files, filesMimes, filesNames):
-        decoded_bytes = b64decode(file)
-        entries = parse_marc(decoded_bytes)
-        file_result[name] = {
-            "mime": mime,
-            "size": naturalsize(len(decoded_bytes)),
-            "entries": entries,
-        }
-        incomplete.insert_incompletes(session, entries)
+        decoded_bytes: bytes = b64decode(file)
+        if mime == "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+            entries = parse_excel(decoded_bytes)
+
+            file_result[name] = {
+                "mime": mime,
+                "size": naturalsize(len(decoded_bytes)),
+                "entries": entries,
+            }
+            incomplete.insert_incompletes(session, entries)
```

Comment:
this chunk of code (from line 46) can be separated out of the `if` clauses

Reply:
Thank you that's a good observation. Will clean it up.